src/transactionsFactories/tokenManagementTransactionsFactory.ts

```diff
@@ -315,12 +341,20 @@ export class TokenManagementTransactionsFactory {
     }): Transaction {
         const dataParts = [
             "setSpecialRole",
-            utf8ToHex(options.tokenIdentifier),
-            addressToHex(options.user),
-            ...(options.addRoleNFTCreate ? [utf8ToHex("ESDTRoleNFTCreate")] : []),
-            ...(options.addRoleNFTBurn ? [utf8ToHex("ESDTRoleNFTBurn")] : []),
-            ...(options.addRoleNFTAddQuantity ? [utf8ToHex("ESDTRoleNFTAddQuantity")] : []),
-            ...(options.addRoleESDTTransferRole ? [utf8ToHex("ESDTTransferRole")] : []),
+            ...this.argSerializer.valuesToStrings([
```

Comment:
A bit hard to read, here / above / below. Previously, code was easier to read.

Maybe do the args trick as in the comment above?

Also:

```
if options.addRoleNFTCreate:
    args.push(new StringValue("ESDTRoleNFTCreate"))
```

That is, ideally, we should only call `valuesToStrings()` once per factory function.

Reply:
Done!